plugin/pkg/global/resourcereferencemanager/admission.go

```diff
@@ -570,3 +629,23 @@ func (r *ReferenceManager) lookupSecret(namespace, name string) error {
 
 	return nil
 }
+
+func (r *ReferenceManager) getAPIResource(groupVersion, kind string) (*metav1.APIResource, error) {
+	resources, err := r.discoveryClient.ServerResourcesForGroupVersion(groupVersion)
```

Comment:
You don't need an extra discovery client for that, it is already part of `kubernetes.Interface`.
Can you remove `r.discoveryClient` and the corresponding methods and fields in the initializer.?
```suggestion
	resources, err := r.kubeClient.Discovery().ServerResourcesForGroupVersion(groupVersion)
```